syft/pkg/dpkg_metadata.go

```diff
@@ -44,3 +46,12 @@ func (m DpkgMetadata) PackageURL(d *distro.Distro) string {
 		"")
 	return pURL.ToString()
 }
+
+func (m DpkgMetadata) ownedFiles() (result []string) {
+	for _, f := range m.Files {
+		if f.Path != "" {
```

Comment:
is it not desirable to prevent duplicates here? Or does it want to intentionally capture everything?

Reply:
I could make this a set implementation and return the resulting slice, think it's worth it?

Reply:
Good question - I'm not sure! But was wondering if it is a problem to have several duplicates here. It does seem that the intent is to report all owned files, not every single file reported from the package owner(s) - in which case it would make sense to prevent duplicates. 

Either a set implementation returning a slice or preventing adding a new item if it already exists in the current slice sounds OK to me

Reply:
Re set implementation — I'm not sure it's worth it.

I do have a question — are we allowing `f.Path` to be an empty string upstream? It'd be nice to be able to rely on a usable value here. I could be misunderstanding how this value gets used

Reply:
Looking at the consumer of this, the idea is to use the owned files to get packages by path. Not de-duplicating here means extra (unnecessary) work for the consumer.

Reply:
re: sets / duplicates: It's not clear if deduplicating signals anything different to the consumer (the given slice, deduplicated or not still indicates ownership), but it does clarify the intent of the interface ("provide a __set__ of file paths that are owned"). I think I'll update to use a set in each, but still return a slice as the return type.

re: empty path: I felt that it was important to ensure that any implementation could not signal "ownership" of "nothing". I didn't check the underlying implementations to see if it was possible, but instead added the defense here. The coincidental implementation of underlying cataloger parsers shouldn't really drive the behavior of "fileOwner" IMHO.